programs/zstdcli.c

```diff
@@ -796,6 +798,57 @@ int main(int argCount, const char* argv[])
                         continue;
                     }
 #endif
+
+                    if (longCommandWArg(&argument, "--filelist=")) {
+                        /* note : in theory, it's better to just store the arguments at this stage,
+                         *    and only start to load & interpret the file after command line is parsed.
+                         *    For a single file, it would be easy to just store its name here, and parse later.
+                         *    However, this implementation makes it possible to read multiple files.
+                         *    An equivalent will have to be able to store multiple file names.
+                         */
+                        FileNamesTable* extendedTable;
+                        FileNamesTable* curTable;
+                        FileNamesTable* concatenatedTables;
+
+                        if (!UTIL_fileExist(argument) || !UTIL_isRegularFile(argument)){
```

Comment:
We already do this check in `UTIL_createFileNamesTable_fromFileName()`. Should we only do it in one place?

Reply:
There is a slight difference.

Here, in case of error, we return `badusage()`, hence it complements the error message with a help screen.

Within `UTIL_createFileNamesTable_fromFileName()`, it just errors out.

`UTIL_createFileNamesTable_fromFileName()` is a library entry point that could be used for other purposes, beyond this sole invocation from `zstdcli.c`. So tests within this library function can't be avoided.

We could remove the tests from `zstdcli.c`, but then we would miss the help screen.